src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/ExtraConfiguration/context/userInputsSchema.js

```diff
@@ -0,0 +1,286 @@
+  boolean: uiBoolean,
+} = USER_INPUT_TYPES
+
+const { array: _array, fixed: _fixed, ...userInputTypes } = USER_INPUT_TYPES
+
+/** @type {UserInputType[]} User inputs types */
+const valuesOfUITypes = Object.values(userInputTypes)
+
+/** @type {Field} Name field */
+const NAME = {
+  name: 'name',
+  label: T.Name,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  disableEnter: true,
+  optionsOnly: false,
+  multiple: false,
+  values: () => useSelector((state) => state.persistent.userInputSuggestionsVR),
+  validation: string()
+    .trim()
+    .required()
+    .default(() => undefined),
+  grid: { sm: 6, md: 4 },
+  fieldProps: { freeSolo: true },
+}
+
+/** @type {Field} Type field */
+const TYPE = {
+  name: 'type',
+  label: T.Type,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  optionsOnly: true,
+  dependOf: NAME.name,
+  values: (name) => {
+    let defaultValues = valuesOfUITypes
+    const sanitizedName = name?.trim()?.toLowerCase()
+    switch (sanitizedName) {
+      case 'memory':
+        defaultValues = [
+          userInputTypes.text,
+          userInputTypes.text64,
+          userInputTypes.number,
+          userInputTypes.range,
+          userInputTypes.list,
+        ]
+        break
+      case 'cpu':
+      case 'vcpu':
+        defaultValues = [
+          userInputTypes.text,
+          userInputTypes.text64,
+          userInputTypes.number,
+          userInputTypes.numberFloat,
+          userInputTypes.range,
+          userInputTypes.rangeFloat,
+          userInputTypes.list,
+        ]
+        break
+      default:
+        break
+    }
+
+    return arrayToOptions(defaultValues, {
+      addEmpty: false,
+      getText: (type) => sentenceCase(type),
+    })
+  },
+  validation: string()
+    .trim()
+    .required()
+    .oneOf(valuesOfUITypes)
+    .default(() => valuesOfUITypes[0]),
+  grid: { sm: 6, md: 4 },
+}
+
+/** @type {Field} Description field */
+const DESCRIPTION = {
+  name: 'label',
+  label: T.Description,
+  type: INPUT_TYPES.TEXT,
+  validation: string()
+    .trim()
+    .notRequired()
+    .default(() => undefined),
+  grid: { sm: 6, md: 4 },
+}
+
+/** @type {Field} Options field */
+const OPTIONS = {
+  name: 'options',
+  label: T.Options,
+  tooltip: [T.PressKeysToAddAValue, ['ENTER']],
+  dependOf: TYPE.name,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  multiple: true,
+  htmlType: (type) =>
+    ![uiList, uiListMultiple].includes(type) && INPUT_TYPES.HIDDEN,
+  validation: array(string().trim())
+    .default(() => [])
+    .when(TYPE.name, (type, schema) =>
+      [uiList, uiListMultiple].includes(type)
+        ? schema.required().min(1)
+        : schema.strip().notRequired()
+    ),
+  fieldProps: {
+    freeSolo: true,
+    placeholder: 'optA,optB,optC',
+  },
+  grid: { md: 8 },
+}
+
+/** @type {{ MIN: Field, MAX: Field }} Range fields */
+const { MIN, MAX } = (() => {
+  const validation = number()
+    .min(0)
+    .default(() => undefined)
+    .when(TYPE.name, (type, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).